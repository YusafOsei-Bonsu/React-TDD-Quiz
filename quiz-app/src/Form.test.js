import React from 'react'
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import Form from './components/form';

describe('Form component', () => {
    let wrapper;

    // Focuses on the Form component for testing
    beforeAll(() => wrapper = shallow(<Form />));

    // Checks if the file exists
    test('File should exist', () => expect(require('./components/form')).toBeTruthy());

    // Form component should render 
    test('Form should render correctly', () => {
        const component = renderer.create(<Form />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    // Checks if the drop-down box renders
    test('Dropdown box should render', () => expect(wrapper.find('select').length).toEqual(1));

    // Checks if the difficulty selectors render
    test('Radiobuttons should render', () => expect(wrapper.find('.levelInputs').length).toEqual(1));

    // Checks if the textbox renders
    test('Textbox should render', () => expect(wrapper.find('#textbox').length).toEqual(1));

    // Wipes the test suite
    afterAll(() => wrapper =null);
});