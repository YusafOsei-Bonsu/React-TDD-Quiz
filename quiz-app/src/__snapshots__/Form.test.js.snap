// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component Form should render correctly 1`] = `
<div
  className="form"
>
  <form>
    <select
      name="topic"
    >
      <option
        value="volvo"
      >
        Volvo
      </option>
      <option
        value="saab"
      >
        Saab
      </option>
      <option
        value="fiat"
      >
        Fiat
      </option>
      <option
        value="audi"
      >
        Audi
      </option>
    </select>
    <br />
    <div
      className="levelInputs"
    >
      <input
        name="difficulty"
        type="radio"
        value="easy"
      />
      <input
        name="difficulty"
        type="radio"
        value="medium"
      />
      <input
        name="difficulty"
        type="radio"
        value="hard"
      />
      <br />
    </div>
    <input
      id="textbox"
      placeholder="player1"
      type="text"
    />
  </form>
</div>
`;
